components/channel_header/channel_header.js

```diff
@@ -462,7 +462,7 @@ class ChannelHeader extends React.PureComponent {
                         ref={this.headerPopoverTextMeasurerRef}
                     >
                         <Markdown
-                            message={headerText.replace(/\n{2,}/g, ' ')}
+                            message={headerText.replace(/\n{2,}/g, ' ').replace(/\n/, ' ')}
```

Comment:
why not changing the first expression instead of replacing twice? also, your regex will only happen on the first line (was that intended?)
```suggestion
                            message={headerText.replace(/\n/g, ' ')}
```
also, should we trim spaces at the begining? like `    \n`, not the most common case, but people tend to hit the spacebar sometimes.

Reply:
First it looks for double newlines and then single newlines, the trick is that it only finds the single newlines after the bigger ones are already replaced. But it might not cover all edge cases. I am really open to suggestions on this.

Reply:
sorry, but I think i'm missing something, what difference would make on the result? although to be fair it would look better if we do it like:
```js
    message={headerText.replace(/\n+/g, ' ')}
```
to prevent having multiple spaces (not that it matters much) but it would be done on a single regex instead of 2

Reply:
I will see how it behaves, makes sense what you are saying :)